### PR TITLE
Polishing

### PR DESCRIPTION
**Pull request by mxmlnkn**
Many bugfixes, tests, benchmarks and other things

### CODE REVIEW

File: examples/threadedExample.cpp [hunk 1]
```diff
@@ -62,25 +66,32 @@ int main( void )
...
+        for( auto i = 1u; i <= 18; i++ )
         {
-            // How about the PNG input? BEWARE! This path is dependent on the folder structure!
-            file = imresh::io::readInFuncs::readPNG( "../examples/testData/imresh.png" );
             // Again, this step is only needed because we have no real images
-            imresh::libs::diffractionIntensity( file.first, file.second );
+            //imresh::libs::diffractionIntensity( file.first, file.second.first, file.second.second );
+            // beware diffraction intensity will be sent to the default stream thereby serializing the addTask calls!!
+            auto tmpTestImage = new float[ imageWidth * imageHeight ];
+            memcpy( tmpTestImage, file.first, imageWidth * imageHeight * sizeof( tmpTestImage[0] ) );
```
Comment (Ferruck) [inline examples/threadedExample.cpp]: What are you doing in this example?

File: src/imresh/io/writeOutFuncs/writeOutFuncs.cpp [hunk 2]
```diff
@@ -26,6 +26,7 @@
 #include <algorithm>
 #ifdef IMRESH_DEBUG
 #   include <iostream>              // std::cout, std::endl
+
```
Comment (Ferruck) [inline src/imresh/io/writeOutFuncs/writeOutFuncs.cpp]: ? :D

File: examples/threadedExample.cpp (hunk 1 above)
Comment (mxmlnkn) [inline examples/threadedExample.cpp]: I'm trying to delete serializing calls in order to get a nice result with nvvp. In this case `diffractionIntensity` is sent to the default stream and for normal input it isn't needed anyway, so I added a diffraction example.

File: src/imresh/io/writeOutFuncs/writeOutFuncs.cpp (hunk 2 above)
Comment (mxmlnkn) [inline src/imresh/io/writeOutFuncs/writeOutFuncs.cpp]: Meh happens. I think I needed iomanip or similar and then deleted my temporary debug code again, but not cleanly

File: examples/threadedExample.cpp (hunk 1 above)
Comment (Ferruck) [inline examples/threadedExample.cpp]: OK.

**Pull request merged**